sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2785,4 +2785,11 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
         "upper" -> size.toString,
         "invalidValue" -> pos.toString))
   }
+
+  def notExistPartitionError(
```

Comment:
shall we put it in `QueryCompilationErrors`?

Reply:
And it's hard to believe we don't have a function to throw `NoSuchPartitionException` already.

Reply:
Okay